[FEATURE] run nox -rs generate in CI

### Is your feature request related to a problem?

The generator may be modifying code when run that corresponds to a change currently being made. 

### What solution would you like?

We want to make sure the generator runs clean as part of every PR. Run `nox -rs generate` in CI and fail if there are more changes to files.

Currently` nox -rs generate` is broken, likely due to recent addition of tests to `nox -rs format`

```
nox -rs generate
nox > Running session generate
nox > Re-using existing virtual environment at .nox/generate.
nox > python -m pip install -rdev-requirements.txt
nox > python utils/generate_api.py
nox > python -m pip install .
nox > python -m pip install black isort
nox > isort setup.py noxfile.py opensearchpy/ test_opensearchpy/ utils/ samples/ benchmarks/ docs/
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/ingest.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/tasks.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/dangling_indices.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/cat.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/security.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/__init__.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/cluster.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/remote_store.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/indices.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/nodes.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/_async/client/snapshot.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/ingest.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/tasks.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/dangling_indices.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/cat.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/security.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/__init__.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/cluster.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/remote_store.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/indices.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/nodes.py
Fixing /Users/saimedhi/opensearchpycode/opensearch-py/opensearchpy/client/snapshot.py
nox > black setup.py noxfile.py opensearchpy/ test_opensearchpy/ utils/ samples/ benchmarks/ docs/
All done! ✨ 🍰 ✨
227 files left unchanged.
nox > python utils/license_headers.py fix setup.py noxfile.py opensearchpy/ test_opensearchpy/ utils/ samples/ benchmarks/ docs/
nox > python -m pip install flake8 black mypy isort pylint types-requests types-six types-simplejson types-python-dateutil types-PyYAML types-mock types-pytz
nox > isort --check setup.py noxfile.py opensearchpy/ test_opensearchpy/ utils/ samples/ benchmarks/ docs/
nox > black --check setup.py noxfile.py opensearchpy/ test_opensearchpy/ utils/ samples/ benchmarks/ docs/
All done! ✨ 🍰 ✨
227 files would be left unchanged.
nox > flake8 setup.py noxfile.py opensearchpy/ test_opensearchpy/ utils/ samples/ benchmarks/ docs/
nox > pylint setup.py noxfile.py opensearchpy/ test_opensearchpy/ utils/ samples/ benchmarks/ docs/

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

nox > python utils/license_headers.py check setup.py noxfile.py opensearchpy/ test_opensearchpy/ utils/ samples/ benchmarks/ docs/
All files had license header
nox > python -m pip install aiohttp
Requirement already satisfied: aiohttp in ./.nox/generate/lib/python3.10/site-packages (3.9.1)
Requirement already satisfied: attrs>=17.3.0 in ./.nox/generate/lib/python3.10/site-packages (from aiohttp) (23.2.0)
Requirement already satisfied: multidict<7.0,>=4.5 in ./.nox/generate/lib/python3.10/site-packages (from aiohttp) (6.0.4)
Requirement already satisfied: yarl<2.0,>=1.0 in ./.nox/generate/lib/python3.10/site-packages (from aiohttp) (1.9.4)
Requirement already satisfied: frozenlist>=1.1.1 in ./.nox/generate/lib/python3.10/site-packages (from aiohttp) (1.4.1)
Requirement already satisfied: aiosignal>=1.1.2 in ./.nox/generate/lib/python3.10/site-packages (from aiohttp) (1.3.1)
Requirement already satisfied: async-timeout<5.0,>=4.0 in ./.nox/generate/lib/python3.10/site-packages (from aiohttp) (4.0.3)
Requirement already satisfied: idna>=2.0 in ./.nox/generate/lib/python3.10/site-packages (from yarl<2.0,>=1.0->aiohttp) (3.6)
nox > mypy --strict setup.py noxfile.py opensearchpy/ test_opensearchpy/ utils/ samples/ benchmarks/ docs/
utils/build_dists.py:47: error: The return type of a generator function should be "Generator" or one of its supertypes  [misc]
test_opensearchpy/test_async/test_http_connection.py:59: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_http_connection.py:91: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_helpers/test_document.py:263: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/test_wrappers.py:35: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/test_wrappers.py:51: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/test_wrappers.py:65: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/test_wrappers.py:80: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/test_wrappers.py:93: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/test_result.py:39: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/test_faceted_search.py:162: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/conftest.py:36: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/conftest.py:46: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/conftest.py:95: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/conftest.py:110: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/test_actions.py:447: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/test_actions.py:680: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/test_actions.py:808: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/test_actions.py:890: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/conftest.py:46: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/conftest.py:86: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_helpers/test_faceted_search.py:69: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_helpers/test_faceted_search.py:94: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_helpers/test_faceted_search.py:115: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_helpers/conftest.py:47: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_helpers/conftest.py:54: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_helpers/conftest.py:64: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_helpers/conftest.py:71: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_helpers/conftest.py:84: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_helpers/conftest.py:108: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_helpers/test_actions.py:121: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_connection.py:346: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server_secured/test_security_plugin.py:13: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server_secured/test_security_plugin.py:25: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/conftest.py:42: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/__init__.py:28: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/__init__.py:36: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_rest_api_spec.py:464: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_server/test_rest_api_spec.py:543: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_helpers/conftest.py:23: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_helpers/conftest.py:33: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_helpers/conftest.py:82: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_helpers/conftest.py:97: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_rest_api_spec.py:212: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_rest_api_spec.py:219: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_helpers/test_faceted_search.py:145: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/test_document.py:252: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/test_faceted_search.py:57: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/test_faceted_search.py:82: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/test_faceted_search.py:103: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/conftest.py:37: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/conftest.py:44: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/conftest.py:54: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/conftest.py:61: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/conftest.py:74: error: Unused "type: ignore" comment  [unused-ignore]
test_opensearchpy/test_async/test_server/test_helpers/conftest.py:98: error: Unused "type: ignore" comment  [unused-ignore]
Found 56 errors in 23 files (checked 227 source files)
nox > Command mypy --strict setup.py noxfile.py opensearchpy/ test_opensearchpy/ utils/ samples/ benchmarks/ docs/ failed with exit code 1
nox > Session generate failed.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -18,6 +18,21 @@ jobs:
           python3.7 -m pip install nox
       - name: Lint the code
         run: nox -s lint
+  
+  generate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.7
+      - name: Install dependencies
+        run: |
+          python3.7 -m pip install nox
+      - name: Run the api generator
+        run: nox -s generate
 
   test-build-distribution:
     runs-on: ubuntu-latest

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Added pylint `pointless-statement` ([#611](https://github.com/opensearch-project/opensearch-py/pull/611))
 - Added a log collection guide ([#579](https://github.com/opensearch-project/opensearch-py/pull/579))
 - Added GHA release ([#614](https://github.com/opensearch-project/opensearch-py/pull/614))
+- Incorporated API generation into CI workflow and fixed 'generate' nox session ([#660](https://github.com/opensearch-project/opensearch-py/pull/660))
 ### Changed
 - Updated the `get_policy` API in the index_management plugin to allow the policy_id argument as optional ([#633](https://github.com/opensearch-project/opensearch-py/pull/633))
 ### Deprecated

--- a/noxfile.py
+++ b/noxfile.py
@@ -74,7 +74,7 @@ def format(session: Any) -> None:
     session.run("black", *SOURCE_FILES)
     session.run("python", "utils/license_headers.py", "fix", *SOURCE_FILES)
 
-    lint(session)
+    session.notify("lint")
 
 
 @nox.session(python=["3.7"])  # type: ignore
@@ -146,4 +146,4 @@ def generate(session: Any) -> None:
     """
     session.install("-rdev-requirements.txt")
     session.run("python", "utils/generate_api.py")
-    format(session)
+    session.notify("format")

--- a/opensearchpy/_async/client/__init__.py
+++ b/opensearchpy/_async/client/__init__.py
@@ -1610,6 +1610,7 @@ class AsyncOpenSearch(Client):
         "rest_total_hits_as_int",
         "routing",
         "scroll",
+        "search_pipeline",
         "search_type",
         "seq_no_primary_term",
         "size",
@@ -1704,6 +1705,8 @@ class AsyncOpenSearch(Client):
         :arg routing: Comma-separated list of specific routing values.
         :arg scroll: Specify how long a consistent view of the index
             should be maintained for scrolled search.
+        :arg search_pipeline: Customizable sequence of processing stages
+            applied to search queries.
         :arg search_type: Search operation type. Valid choices are
             query_then_fetch, dfs_query_then_fetch.
         :arg seq_no_primary_term: Specify whether to return sequence

--- a/opensearchpy/_async/client/cat.py
+++ b/opensearchpy/_async/client/cat.py
@@ -70,7 +70,7 @@ class CatClient(NamespacedClient):
             "GET", _make_path("_cat", "aliases", name), params=params, headers=headers
         )
 
-    @query_params()
+    @query_params("bytes", "format", "h", "help", "s", "v")
     async def all_pit_segments(
         self,
         params: Any = None,
@@ -79,6 +79,16 @@ class CatClient(NamespacedClient):
         """
         Lists all active point-in-time segments.
 
+
+        :arg bytes: The unit in which to display byte values. Valid
+            choices are b, k, kb, m, mb, g, gb, t, tb, p, pb.
+        :arg format: A short version of the Accept header, e.g. json,
+            yaml.
+        :arg h: Comma-separated list of column names to display.
+        :arg help: Return help information. Default is false.
+        :arg s: Comma-separated list of column names or column aliases
+            to sort by.
+        :arg v: Verbose mode. Display column headers. Default is false.
         """
         return await self.transport.perform_request(
             "GET", "/_cat/pit_segments/_all", params=params, headers=headers
@@ -503,7 +513,7 @@ class CatClient(NamespacedClient):
             "GET", "/_cat/pending_tasks", params=params, headers=headers
         )
 
-    @query_params()
+    @query_params("bytes", "format", "h", "help", "s", "v")
     async def pit_segments(
         self,
         body: Any = None,
@@ -514,6 +524,15 @@ class CatClient(NamespacedClient):
         List segments for one or several PITs.
 
 
+        :arg bytes: The unit in which to display byte values. Valid
+            choices are b, k, kb, m, mb, g, gb, t, tb, p, pb.
+        :arg format: A short version of the Accept header, e.g. json,
+            yaml.
+        :arg h: Comma-separated list of column names to display.
+        :arg help: Return help information. Default is false.
+        :arg s: Comma-separated list of column names or column aliases
+            to sort by.
+        :arg v: Verbose mode. Display column headers. Default is false.
         """
         return await self.transport.perform_request(
             "GET", "/_cat/pit_segments", params=params, headers=headers, body=body
@@ -632,15 +651,20 @@ class CatClient(NamespacedClient):
 
     @query_params(
         "active_only",
+        "allow_no_indices",
         "bytes",
         "completed_only",
         "detailed",
+        "expand_wildcards",
         "format",
         "h",
         "help",
+        "ignore_throttled",
+        "ignore_unavailable",
         "s",
         "shards",
         "time",
+        "timeout",
         "v",
     )
     async def segment_replication(
@@ -658,21 +682,32 @@ class CatClient(NamespacedClient):
             names to limit the returned information.
         :arg active_only: If `true`, the response only includes ongoing
             segment replication events. Default is false.
+        :arg allow_no_indices: Whether to ignore if a wildcard indices
+            expression resolves into no concrete indices. (This includes `_all`
+            string or when no indices have been specified).
         :arg bytes: The unit in which to display byte values. Valid
             choices are b, k, kb, m, mb, g, gb, t, tb, p, pb.
         :arg completed_only: If `true`, the response only includes
             latest completed segment replication events. Default is false.
         :arg detailed: If `true`, the response includes detailed
             information about segment replications. Default is false.
+        :arg expand_wildcards: Whether to expand wildcard expression to
+            concrete indices that are open, closed or both. Valid choices are all,
+            open, closed, hidden, none.
         :arg format: A short version of the Accept header, e.g. json,
             yaml.
         :arg h: Comma-separated list of column names to display.
         :arg help: Return help information. Default is false.
+        :arg ignore_throttled: Whether specified concrete, expanded or
+            aliased indices should be ignored when throttled.
+        :arg ignore_unavailable: Whether specified concrete indices
+            should be ignored when unavailable (missing or closed).
         :arg s: Comma-separated list of column names or column aliases
             to sort by.
         :arg shards: Comma-separated list of shards to display.
         :arg time: The unit in which to display time values. Valid
             choices are d, h, m, s, ms, micros, nanos.
+        :arg timeout: Operation timeout.
         :arg v: Verbose mode. Display column headers. Default is false.
         """
         return await self.transport.perform_request(

--- a/opensearchpy/_async/client/indices.py
+++ b/opensearchpy/_async/client/indices.py
@@ -166,7 +166,12 @@ class IndicesClient(NamespacedClient):
         )
 
     @query_params(
-        "cluster_manager_timeout", "master_timeout", "timeout", "wait_for_active_shards"
+        "cluster_manager_timeout",
+        "master_timeout",
+        "task_execution_timeout",
+        "timeout",
+        "wait_for_active_shards",
+        "wait_for_completion",
     )
     async def clone(
         self,
@@ -189,9 +194,13 @@ class IndicesClient(NamespacedClient):
         :arg master_timeout (Deprecated: To promote inclusive language,
             use 'cluster_manager_timeout' instead.): Operation timeout for
             connection to master node.
+        :arg task_execution_timeout: Explicit task execution timeout,
+            only useful when wait_for_completion is false, defaults to 1h.
         :arg timeout: Operation timeout.
         :arg wait_for_active_shards: Set the number of active shards to
             wait for on the cloned index before the operation returns.
+        :arg wait_for_completion: Should this request wait until the
+            operation has completed before returning. Default is True.
         """
         for param in (index, target):
             if param in SKIP_IN_PATH:
@@ -260,8 +269,10 @@ class IndicesClient(NamespacedClient):
         "expand_wildcards",
         "ignore_unavailable",
         "master_timeout",
+        "task_execution_timeout",
         "timeout",
         "wait_for_active_shards",
+        "wait_for_completion",
     )
     async def open(
         self,
@@ -287,9 +298,13 @@ class IndicesClient(NamespacedClient):
         :arg master_timeout (Deprecated: To promote inclusive language,
             use 'cluster_manager_timeout' instead.): Operation timeout for
             connection to master node.
+        :arg task_execution_timeout: Explicit task execution timeout,
+            only useful when wait_for_completion is false, defaults to 1h.
         :arg timeout: Operation timeout.
         :arg wait_for_active_shards: Sets the number of active shards to
             wait for before the operation returns.
+        :arg wait_for_completion: Should this request wait until the
+            operation has completed before returning. Default is True.
         """
         if index in SKIP_IN_PATH:
             raise ValueError("Empty value passed for a required argument 'index'.")
@@ -1245,6 +1260,7 @@ class IndicesClient(NamespacedClient):
         "ignore_unavailable",
         "max_num_segments",
         "only_expunge_deletes",
+        "wait_for_completion",
     )
     async def forcemerge(
         self,
@@ -1272,6 +1288,8 @@ class IndicesClient(NamespacedClient):
             be merged into (default: dynamic).
         :arg only_expunge_deletes: Specify whether the operation should
             only expunge deleted documents.
+        :arg wait_for_completion: Should this request wait until the
+            operation has completed before returning. Default is True.
         """
         return await self.transport.perform_request(
             "POST", _make_path(index, "_forcemerge"), params=params, headers=headers
@@ -1281,8 +1299,10 @@ class IndicesClient(NamespacedClient):
         "cluster_manager_timeout",
         "copy_settings",
         "master_timeout",
+        "task_execution_timeout",
         "timeout",
         "wait_for_active_shards",
+        "wait_for_completion",
     )
     async def shrink(
         self,
@@ -1307,9 +1327,13 @@ class IndicesClient(NamespacedClient):
         :arg master_timeout (Deprecated: To promote inclusive language,
             use 'cluster_manager_timeout' instead.): Operation timeout for
             connection to master node.
+        :arg task_execution_timeout: Explicit task execution timeout,
+            only useful when wait_for_completion is false, defaults to 1h.
         :arg timeout: Operation timeout.
         :arg wait_for_active_shards: Set the number of active shards to
             wait for on the shrunken index before the operation returns.
+        :arg wait_for_completion: Should this request wait until the
+            operation has completed before returning. Default is True.
         """
         for param in (index, target):
             if param in SKIP_IN_PATH:
@@ -1327,8 +1351,10 @@ class IndicesClient(NamespacedClient):
         "cluster_manager_timeout",
         "copy_settings",
         "master_timeout",
+        "task_execution_timeout",
         "timeout",
         "wait_for_active_shards",
+        "wait_for_completion",
     )
     async def split(
         self,
@@ -1354,9 +1380,13 @@ class IndicesClient(NamespacedClient):
         :arg master_timeout (Deprecated: To promote inclusive language,
             use 'cluster_manager_timeout' instead.): Operation timeout for
             connection to master node.
+        :arg task_execution_timeout: Explicit task execution timeout,
+            only useful when wait_for_completion is false, defaults to 1h.
         :arg timeout: Operation timeout.
         :arg wait_for_active_shards: Set the number of active shards to
             wait for on the shrunken index before the operation returns.
+        :arg wait_for_completion: Should this request wait until the
+            operation has completed before returning. Default is True.
         """
         for param in (index, target):
             if param in SKIP_IN_PATH:

--- a/opensearchpy/_async/client/nodes.py
+++ b/opensearchpy/_async/client/nodes.py
@@ -122,7 +122,8 @@ class NodesClient(NamespacedClient):
             nodes.
         :arg metric: Limit the information returned to the specified
             metrics. Valid choices are _all, breaker, fs, http, indices, jvm, os,
-            process, thread_pool, transport, discovery, indexing_pressure.
+            process, thread_pool, transport, discovery, indexing_pressure,
+            search_pipeline.
         :arg index_metric: Limit the information returned for `indices`
             metric to the specific index metrics. Isn't used if `indices` (or `all`)
             metric isn't specified. Valid choices are _all, store, indexing, get,

--- a/opensearchpy/client/__init__.py
+++ b/opensearchpy/client/__init__.py
@@ -1610,6 +1610,7 @@ class OpenSearch(Client):
         "rest_total_hits_as_int",
         "routing",
         "scroll",
+        "search_pipeline",
         "search_type",
         "seq_no_primary_term",
         "size",
@@ -1704,6 +1705,8 @@ class OpenSearch(Client):
         :arg routing: Comma-separated list of specific routing values.
         :arg scroll: Specify how long a consistent view of the index
             should be maintained for scrolled search.
+        :arg search_pipeline: Customizable sequence of processing stages
+            applied to search queries.
         :arg search_type: Search operation type. Valid choices are
             query_then_fetch, dfs_query_then_fetch.
         :arg seq_no_primary_term: Specify whether to return sequence

--- a/opensearchpy/client/cat.py
+++ b/opensearchpy/client/cat.py
@@ -70,7 +70,7 @@ class CatClient(NamespacedClient):
             "GET", _make_path("_cat", "aliases", name), params=params, headers=headers
         )
 
-    @query_params()
+    @query_params("bytes", "format", "h", "help", "s", "v")
     def all_pit_segments(
         self,
         params: Any = None,
@@ -79,6 +79,16 @@ class CatClient(NamespacedClient):
         """
         Lists all active point-in-time segments.
 
+
+        :arg bytes: The unit in which to display byte values. Valid
+            choices are b, k, kb, m, mb, g, gb, t, tb, p, pb.
+        :arg format: A short version of the Accept header, e.g. json,
+            yaml.
+        :arg h: Comma-separated list of column names to display.
+        :arg help: Return help information. Default is false.
+        :arg s: Comma-separated list of column names or column aliases
+            to sort by.
+        :arg v: Verbose mode. Display column headers. Default is false.
         """
         return self.transport.perform_request(
             "GET", "/_cat/pit_segments/_all", params=params, headers=headers
@@ -503,7 +513,7 @@ class CatClient(NamespacedClient):
             "GET", "/_cat/pending_tasks", params=params, headers=headers
         )
 
-    @query_params()
+    @query_params("bytes", "format", "h", "help", "s", "v")
     def pit_segments(
         self,
         body: Any = None,
@@ -514,6 +524,15 @@ class CatClient(NamespacedClient):
         List segments for one or several PITs.
 
 
+        :arg bytes: The unit in which to display byte values. Valid
+            choices are b, k, kb, m, mb, g, gb, t, tb, p, pb.
+        :arg format: A short version of the Accept header, e.g. json,
+            yaml.
+        :arg h: Comma-separated list of column names to display.
+        :arg help: Return help information. Default is false.
+        :arg s: Comma-separated list of column names or column aliases
+            to sort by.
+        :arg v: Verbose mode. Display column headers. Default is false.
         """
         return self.transport.perform_request(
             "GET", "/_cat/pit_segments", params=params, headers=headers, body=body
@@ -632,15 +651,20 @@ class CatClient(NamespacedClient):
 
     @query_params(
         "active_only",
+        "allow_no_indices",
         "bytes",
         "completed_only",
         "detailed",
+        "expand_wildcards",
         "format",
         "h",
         "help",
+        "ignore_throttled",
+        "ignore_unavailable",
         "s",
         "shards",
         "time",
+        "timeout",
         "v",
     )
     def segment_replication(
@@ -658,21 +682,32 @@ class CatClient(NamespacedClient):
             names to limit the returned information.
         :arg active_only: If `true`, the response only includes ongoing
             segment replication events. Default is false.
+        :arg allow_no_indices: Whether to ignore if a wildcard indices
+            expression resolves into no concrete indices. (This includes `_all`
+            string or when no indices have been specified).
         :arg bytes: The unit in which to display byte values. Valid
             choices are b, k, kb, m, mb, g, gb, t, tb, p, pb.
         :arg completed_only: If `true`, the response only includes
             latest completed segment replication events. Default is false.
         :arg detailed: If `true`, the response includes detailed
             information about segment replications. Default is false.
+        :arg expand_wildcards: Whether to expand wildcard expression to
+            concrete indices that are open, closed or both. Valid choices are all,
+            open, closed, hidden, none.
         :arg format: A short version of the Accept header, e.g. json,
             yaml.
         :arg h: Comma-separated list of column names to display.
         :arg help: Return help information. Default is false.
+        :arg ignore_throttled: Whether specified concrete, expanded or
+            aliased indices should be ignored when throttled.
+        :arg ignore_unavailable: Whether specified concrete indices
+            should be ignored when unavailable (missing or closed).
         :arg s: Comma-separated list of column names or column aliases
             to sort by.
         :arg shards: Comma-separated list of shards to display.
         :arg time: The unit in which to display time values. Valid
             choices are d, h, m, s, ms, micros, nanos.
+        :arg timeout: Operation timeout.
         :arg v: Verbose mode. Display column headers. Default is false.
         """
         return self.transport.perform_request(

--- a/opensearchpy/client/indices.py
+++ b/opensearchpy/client/indices.py
@@ -166,7 +166,12 @@ class IndicesClient(NamespacedClient):
         )
 
     @query_params(
-        "cluster_manager_timeout", "master_timeout", "timeout", "wait_for_active_shards"
+        "cluster_manager_timeout",
+        "master_timeout",
+        "task_execution_timeout",
+        "timeout",
+        "wait_for_active_shards",
+        "wait_for_completion",
     )
     def clone(
         self,
@@ -189,9 +194,13 @@ class IndicesClient(NamespacedClient):
         :arg master_timeout (Deprecated: To promote inclusive language,
             use 'cluster_manager_timeout' instead.): Operation timeout for
             connection to master node.
+        :arg task_execution_timeout: Explicit task execution timeout,
+            only useful when wait_for_completion is false, defaults to 1h.
         :arg timeout: Operation timeout.
         :arg wait_for_active_shards: Set the number of active shards to
             wait for on the cloned index before the operation returns.
+        :arg wait_for_completion: Should this request wait until the
+            operation has completed before returning. Default is True.
         """
         for param in (index, target):
             if param in SKIP_IN_PATH:
@@ -260,8 +269,10 @@ class IndicesClient(NamespacedClient):
         "expand_wildcards",
         "ignore_unavailable",
         "master_timeout",
+        "task_execution_timeout",
         "timeout",
         "wait_for_active_shards",
+        "wait_for_completion",
     )
     def open(
         self,
@@ -287,9 +298,13 @@ class IndicesClient(NamespacedClient):
         :arg master_timeout (Deprecated: To promote inclusive language,
             use 'cluster_manager_timeout' instead.): Operation timeout for
             connection to master node.
+        :arg task_execution_timeout: Explicit task execution timeout,
+            only useful when wait_for_completion is false, defaults to 1h.
         :arg timeout: Operation timeout.
         :arg wait_for_active_shards: Sets the number of active shards to
             wait for before the operation returns.
+        :arg wait_for_completion: Should this request wait until the
+            operation has completed before returning. Default is True.
         """
         if index in SKIP_IN_PATH:
             raise ValueError("Empty value passed for a required argument 'index'.")
@@ -1245,6 +1260,7 @@ class IndicesClient(NamespacedClient):
         "ignore_unavailable",
         "max_num_segments",
         "only_expunge_deletes",
+        "wait_for_completion",
     )
     def forcemerge(
         self,
@@ -1272,6 +1288,8 @@ class IndicesClient(NamespacedClient):
             be merged into (default: dynamic).
         :arg only_expunge_deletes: Specify whether the operation should
             only expunge deleted documents.
+        :arg wait_for_completion: Should this request wait until the
+            operation has completed before returning. Default is True.
         """
         return self.transport.perform_request(
             "POST", _make_path(index, "_forcemerge"), params=params, headers=headers
@@ -1281,8 +1299,10 @@ class IndicesClient(NamespacedClient):
         "cluster_manager_timeout",
         "copy_settings",
         "master_timeout",
+        "task_execution_timeout",
         "timeout",
         "wait_for_active_shards",
+        "wait_for_completion",
     )
     def shrink(
         self,
@@ -1307,9 +1327,13 @@ class IndicesClient(NamespacedClient):
         :arg master_timeout (Deprecated: To promote inclusive language,
             use 'cluster_manager_timeout' instead.): Operation timeout for
             connection to master node.
+        :arg task_execution_timeout: Explicit task execution timeout,
+            only useful when wait_for_completion is false, defaults to 1h.
         :arg timeout: Operation timeout.
         :arg wait_for_active_shards: Set the number of active shards to
             wait for on the shrunken index before the operation returns.
+        :arg wait_for_completion: Should this request wait until the
+            operation has completed before returning. Default is True.
         """
         for param in (index, target):
             if param in SKIP_IN_PATH:
@@ -1327,8 +1351,10 @@ class IndicesClient(NamespacedClient):
         "cluster_manager_timeout",
         "copy_settings",
         "master_timeout",
+        "task_execution_timeout",
         "timeout",
         "wait_for_active_shards",
+        "wait_for_completion",
     )
     def split(
         self,
@@ -1354,9 +1380,13 @@ class IndicesClient(NamespacedClient):
         :arg master_timeout (Deprecated: To promote inclusive language,
             use 'cluster_manager_timeout' instead.): Operation timeout for
             connection to master node.
+        :arg task_execution_timeout: Explicit task execution timeout,
+            only useful when wait_for_completion is false, defaults to 1h.
         :arg timeout: Operation timeout.
         :arg wait_for_active_shards: Set the number of active shards to
             wait for on the shrunken index before the operation returns.
+        :arg wait_for_completion: Should this request wait until the
+            operation has completed before returning. Default is True.
         """
         for param in (index, target):
             if param in SKIP_IN_PATH:

--- a/opensearchpy/client/nodes.py
+++ b/opensearchpy/client/nodes.py
@@ -122,7 +122,8 @@ class NodesClient(NamespacedClient):
             nodes.
         :arg metric: Limit the information returned to the specified
             metrics. Valid choices are _all, breaker, fs, http, indices, jvm, os,
-            process, thread_pool, transport, discovery, indexing_pressure.
+            process, thread_pool, transport, discovery, indexing_pressure,
+            search_pipeline.
         :arg index_metric: Limit the information returned for `indices`
             metric to the specific index metrics. Isn't used if `indices` (or `all`)
             metric isn't specified. Valid choices are _all, store, indexing, get,

